app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -10,14 +16,40 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
         val colorCircleView = findViewById<SelectedColorView>(R.id.colorCircle)
         val drawingFieldView = findViewById<DrawingFieldView>(R.id.drawField)
+        findViewById<TextView>(R.id.newLayer).setOnClickListener {
+            layerAlertDialog()
+        }
 
         colorCircleView.onColorChange = { stateColor ->
             model.primaryColor = stateColor
             drawingFieldView.updateColor(stateColor)
         }
 
     }
+
+    fun layerAlertDialog() {
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        layerField.setText(getString(R.string.new_layer_hint, model.layers.size + 1))
+
+        val dialog = AlertDialog.Builder(this).apply {
+            setView(view)
+        }.create()
+
+        view.findViewById<TextView>(R.id.newLayerCancel).setOnClickListener {
+            dialog.dismiss()
+        }
+        view.findViewById<TextView>(R.id.newLayerSubmit).setOnClickListener {
+            //adding empty bitmaps for the time being so we can increment the layer list
+            model.layers.add(Layer(layerField.text.toString(),Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)))
```

Comment:
You shouldn't be making this on the main thread so nullable may be more appropriate, but regardless this is outside ticket scope, please remove

Reply:
I removed it